sdk/helper/clientcountutil/clientcountutil.go

```diff
@@ -0,0 +1,385 @@
+func NewActivityLogData(client *api.Client) *ActivityLogDataGenerator {
+	return &ActivityLogDataGenerator{
+		client: client,
+		data:   new(generation.ActivityLogMockInput),
+	}
+}
+
+// NewCurrentMonthData opens a new month of data for the current month. All
+// clients will continue to be added to this month until a new month is created
+// with NewPreviousMonthData.
+func (d *ActivityLogDataGenerator) NewCurrentMonthData() *ActivityLogDataGenerator {
+	d.newMonth(&generation.Data{Month: &generation.Data_CurrentMonth{CurrentMonth: true}})
+	return d
+}
+
+// NewPreviousMonthData opens a new month of data, where the clients will be
+// recorded as having been seen monthsAgo months ago. All clients will continue
+// to be added to this month until a new month is created with
+// NewPreviousMonthData or NewCurrentMonthData.
+func (d *ActivityLogDataGenerator) NewPreviousMonthData(monthsAgo int) *ActivityLogDataGenerator {
+	return d.newMonth(&generation.Data{Month: &generation.Data_MonthsAgo{MonthsAgo: int32(monthsAgo)}})
+}
+
+func (d *ActivityLogDataGenerator) newMonth(newMonth *generation.Data) *ActivityLogDataGenerator {
+	d.data.Data = append(d.data.Data, newMonth)
+	d.addingToMonth = newMonth
+	d.addingToSegment = nil
+	return d
+}
+
+// MonthOption holds an option that can be set for the entire month
+type MonthOption func(m *generation.Data)
+
+// WithMaximumSegmentIndex sets the maximum segment index for the segments in
+// the open month. Set this value in order to set how many indexes the data
+// should be split across. This must include any empty or skipped indexes. For
+// example, say that you would like all of your data split across indexes 0 and
+// 3, with the following empty and skipped indexes:
+//
+//	empty indexes: [2]
+//	skipped indexes: [1]
+//
+// To accomplish that, you will need to call WithMaximumSegmentIndex(3).
+// This value will be ignored if you have called Segment() for the open month
+// If not set, all data will be in 1 segment.
+func WithMaximumSegmentIndex(n int) MonthOption {
+	return func(m *generation.Data) {
+		m.NumSegments = int32(n)
+	}
+}
+
+// WithEmptySegmentIndexes sets which segment indexes should be empty for the
+// segments in the open month. If you use this option, you must either:
+//  1. ensure that you've called Segment() for the open month
+//  2. use WithMaximumSegmentIndex() to set the total number of segments
+func WithEmptySegmentIndexes(i ...int) MonthOption {
+	return func(m *generation.Data) {
+		indexes := make([]int32, 0, len(i))
+		for _, index := range i {
+			indexes = append(indexes, int32(index))
+		}
+		m.EmptySegmentIndexes = indexes
+	}
+}
+
+// WithSkipSegmentIndexes sets which segment indexes should be skipped for the
+// segments in the open month. If you use this option, you must either:
+//  1. ensure that you've called Segment() for the open month
+//  2. use WithMaximumSegmentIndex() to set the total number of segments
```

Comment:
I appreciate these comments that spell out what needs to happen in order to use the library successfully. What happens if someone doesn't follow these steps when calling these functions?

Reply:
added a description of the implications